### DISCUSSION TITLE
Architecture File Transfer

### DISCUSSION THREAD

**Discussion by MaliaCataleya** [eclipse-edc/Connector#2503] [Q&A]
Hey :)
I'm having trouble understanding how I should build an architecture to enable file transfer in a network.

How I understood it so far is that I basically need a launcher (build file + Runtime). Then I'd be able to add functionality via extensions.
But looking at the Connector Repo I still don't understand which files I need to copy to my workspace and how I should import them (due to the variety of folders and files).

So far I've worked through the Samples directory (and watched the onboarding series) and tried to change the transfer-02 so that I can exchange data from one client to another over the same network. While the consumer on the other client (other IP) noticed that it was supposed to initiate the contract negotiation (after sending the curl command to the client), it still tried to call/ reference the localhost provider.
Since I couldn't fix this error I figured to implement it the 'correct' way using the Connector Repo, but then my initial question arose on how I'm actually supposed to implement it (basically what exact files do I need).

As you can probably see, I'm pretty confused :D 
It'd be great if someone would help me figure this out.

TL;DR:
1. Can I change the Sample transfer-02 so that file exchange from on client to another (in the same network) is possible?
2. If not, which files from the Connector Repo do I need to fulfill this usecase?

Thanks in advance!
